Simperium/SPPersistentMutableDictionary.m

```diff
@@ -450,15 +462,31 @@ - (NSArray *)loadObjectsProperty:(NSString*)property unarchive:(BOOL)unarchive {
                 continue;
             }
             
-            if (unarchive) {
-                [keys addObject:[NSKeyedUnarchiver unarchiveObjectWithData:value]];
-            } else {
-                [keys addObject:value];
+            if (!unarchive) {
+                [output addObject:value];
+                continue;;
+            }
+
+            id decodedValue = [NSKeyedUnarchiver unarchivedObjectOfClasses:self.supportedObjectTypes
+                                                                  fromData:value
+                                                                     error:nil];
+            if (decodedValue) {
+                [output addObject:decodedValue];
             }
         }
     }];
     
-    return keys;
+    return output;
+}
+
+- (BOOL)canStoreObject:(id)anObject {
+    for (Class supportedClass in self.supportedObjectTypes) {
+        if ([anObject isKindOfClass:supportedClass]) {
+            return YES;
+        }
+    }
+
+    return NO;
 }
```

Comment:
Do we need to check recursively in case of NSArray / NSDictionary?

Reply:
Reasoning behind that assertion is: ObjC doesn't really have Generics, but "Lightweight Generics".

You can set a `__covariant` type in the header file, but it's only for Swift bridging. So I tried to do something that, at least, would give you a warning.

In our specific use case, all of the dictionaries contain either Strings or Numbers (everything is generated by a single API), so no need to validate further).

**Plus** the serialization invocation itself would blow up, if you pass along a non NSSecureCoding compliant class.

**Thank you** for bringing that up!!